src/textual/message_pump.py

```diff
@@ -374,6 +374,7 @@ def call_next(self, callback: Callable, *args, **kwargs) -> None:
             **kwargs: Keyword arguments to pass to the callable.
         """
         self._next_callbacks.append(partial(callback, *args, **kwargs))
+        self.check_idle()
```

Comment:
A bug! Without this, the next callbacks wouldn't run outside of a different message handler.